config/bootstrap.php

```diff
@@ -23,16 +23,48 @@
 
 // Load cached env vars if the .env.local.php file exists
 // Run "composer dump-env prod" to create it (requires symfony/flex >=1.2)
-if (is_array($env = @include dirname(__DIR__) . '/.env.local.php')) {
-    $_SERVER += $env;
-    $_ENV += $env;
+if (is_array($env = @include dirname(__DIR__) . '/.env.local.php') && ($_SERVER['APP_ENV'] ?? $_ENV['APP_ENV'] ?? $env['APP_ENV']) === $env['APP_ENV']) {
+    foreach ($env as $k => $v) {
+        $_ENV[$k] = $_ENV[$k] ?? (isset($_SERVER[$k]) && 0 !== strpos($k, 'HTTP_') ? $_SERVER[$k] : $v);
+    }
 } elseif (!class_exists(Dotenv::class)) {
     throw new RuntimeException('Please run "composer require symfony/dotenv" to load the ".env" files configuring the application.');
 } else {
+    $path = dirname(__DIR__) . '/.env';
+    $dotenv = new Dotenv(false);
```

Comment:
do you `new Dotenv(false)` on purpose ? if you do so, you may experience difficulties with system variables overrides.

Reply:
Beware that `putenv()` is not thread safe and this setting will default to `false` in Symfony 5.0.

Reply:
i'm aware, just worries it wont break your deployed simple roster instances :)